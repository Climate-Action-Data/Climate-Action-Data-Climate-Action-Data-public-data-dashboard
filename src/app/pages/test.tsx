'use client'
import { type NextPage } from "next";

const Test: NextPage = () => {
    return (
        <div>
            <h1>Hello</h1>
            <p>data</p>
        </div>
    )
}

export default Test