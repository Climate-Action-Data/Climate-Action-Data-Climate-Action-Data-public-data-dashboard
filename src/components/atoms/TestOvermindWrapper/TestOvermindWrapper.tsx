import { createOvermind } from 'overmind'
import { Provider } from 'overmind-react'
import { config } from '@/overmind'
import { PropsWithChildren } from 'react'

const overmind = createOvermind(config)

export const TestOvermindWrapper = (props: PropsWithChildren) => {
  return <Provider value={overmind}>{props.children}</Provider>
}
