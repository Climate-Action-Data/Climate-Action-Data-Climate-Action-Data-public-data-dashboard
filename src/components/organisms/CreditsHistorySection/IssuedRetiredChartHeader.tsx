import { Grid, GridItem, HStack, Tag, TagCloseButton, TagLabel } from '@chakra-ui/react'
import SubregionIndicator from '@/components/atoms/SubregionIndicator/SubregionIndicator'
import { SubRegion } from '@/@types/geojson'
import { AutoComplete } from '@/components/molecules/AutoComplete/AutoComplete'
import { TimeframesData } from '@/@types/Timeframe'
import { useActions, useAppState } from '@/overmind'
import { useTranslation } from 'react-i18next'
import { FC } from 'react'

interface IssuedRetiredChartHeaderProps {
  autocompleteItems: { value: string; label: string }[]
}

const IssuedRetiredChartHeader: FC<IssuedRetiredChartHeaderProps> = (props) => {
  const { setCountry, setSubRegion, setTimeframe } = useActions().creditsHistory
  const { dataFilters } = useAppState().creditsHistory
  const { t } = useTranslation(`home`)
  const { t: countryTranslate } = useTranslation(`countries`)

  const { autocompleteItems } = props

  const getCountryPlaceholder = (country: string | undefined): string => {
    if (country) {
      return countryTranslate(`${country}`)
    }
    return t(`regions.chooseCountry`)
  }

  return (
    <Grid
      id={`issued-retired-chart-header`}
      templateAreas={`
              'title continents-filter'
              'date-range-filter date-range-filter'
            `}
      gridTemplateColumns={`auto 1fr`}
      rowGap={`16px`}
      alignItems={`center`}
    >
      <GridItem area={`title`} borderRight={`1px`} borderRightColor={`lightGray.400`} padding={`8px 16px `}>
        <SubregionIndicator subregion={dataFilters?.region} clearSubregion={() => setSubRegion(SubRegion.WORLD)} />
      </GridItem>
      <GridItem area={`continents-filter`} alignSelf={`center`} padding={`4px `}>
        <HStack padding={`4px 16px`} gap={`8px`} alignContent={`baseline`}>
          {dataFilters.region === SubRegion.WORLD ? (
            <AutoComplete
              onItemClick={(region) => setSubRegion(region.value)}
              onItemHover={(_) => undefined}
              onDropDownLeave={() => undefined}
              items={autocompleteItems}
              placeholder={t(`regions.chooseRegion`)}
            />
          ) : (
            <AutoComplete
              onItemClick={(country) => setCountry(country.value)}
              onItemHover={(_) => undefined}
              onDropDownLeave={() => undefined}
              items={autocompleteItems}
              placeholder={getCountryPlaceholder(dataFilters.country)}
            />
          )}
        </HStack>
      </GridItem>
      <GridItem area={`date-range-filter`}>
        <HStack padding={`4px 16px`} gap={`8px`}>
          {Object.values(TimeframesData).map(
            (timeframe, idx) =>
              timeframe !== TimeframesData.MAX && (
                <Tag
                  size={`md`}
                  key={idx}
                  borderRadius="full"
                  _hover={{ cursor: `pointer` }}
                  variant={dataFilters.timeframe === timeframe ? `solid` : `outline`}
                  colorScheme="gray"
                >
                  <TagLabel data-testid={`button-timeframe-${idx}`} onClick={() => setTimeframe(timeframe)}>
                    {t(`timeframes.${timeframe}`)}
                  </TagLabel>
                  {dataFilters.timeframe === timeframe && <TagCloseButton data-testid={`button-timeframe-close`} onClick={() => setTimeframe(TimeframesData.MAX)} />}
                </Tag>
              ),
          )}
        </HStack>
      </GridItem>
    </Grid>
  )
}

export default IssuedRetiredChartHeader
