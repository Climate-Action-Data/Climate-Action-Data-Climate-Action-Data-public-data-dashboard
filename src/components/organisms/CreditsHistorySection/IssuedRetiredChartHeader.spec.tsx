import { render } from '@testing-library/react'

import { TestOvermindWrapper } from '@/test/TestOvermindWrapper'
import IssuedRetiredChartHeader from '@/components/organisms/CreditsHistorySection/IssuedRetiredChartHeader'

test(`renders correctly`, () => {
  const { container } = render(
    <TestOvermindWrapper>
      <IssuedRetiredChartHeader autocompleteItems={[]} />
    </TestOvermindWrapper>,
  )
  expect(container).toMatchSnapshot()
})
