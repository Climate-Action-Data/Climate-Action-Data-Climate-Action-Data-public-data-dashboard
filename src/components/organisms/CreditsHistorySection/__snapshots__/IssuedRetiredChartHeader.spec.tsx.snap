// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="css-12rn99f"
    id="issued-retired-chart-header"
  >
    <div
      class="css-a3yeag"
    >
      <div
        class="chakra-stack css-1igwmid"
      >
        <p
          class="chakra-text className css-v7dfld"
          style="font-family: fontFamily;"
          variable="variable"
        >
          World
        </p>
      </div>
    </div>
    <div
      class="css-1q2dfsb"
    >
      <div
        class="chakra-stack css-s7zwns"
      >
        <div
          class="chakra-stack css-8g8ihq"
        >
          <div
            aria-controls="popover-content-:r1:"
            aria-expanded="false"
            aria-haspopup="dialog"
            class="chakra-stack css-8g8ihq"
            id="popover-trigger-:r1:"
          >
            <div
              class="chakra-input__group css-bx0blc"
              data-group="true"
            >
              <input
                class="chakra-input css-1luyivc"
                data-testid="dropdown_input"
                placeholder="Choose a region"
              />
              <div
                class="chakra-input__right-element css-2cnd2z"
              >
                <button
                  aria-label="toggle menu"
                  class="chakra-button css-xsdlqf"
                  data-testid="dropdown-button"
                  type="button"
                >
                  <svg
                    class="chakra-icon css-c0air1"
                    focusable="false"
                    viewBox="0 0 8 4"
                  >
                    <path
                      d="M3.99911 3.9034L0.191406 0.0957031H7.80681L3.99911 3.9034Z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
          <div
            class="chakra-popover__popper css-0"
            style="visibility: hidden; position: absolute; inset: 0 auto auto 0;"
          >
            <section
              aria-describedby="popover-body-:r1:"
              class="chakra-popover__content css-18u085x"
              data-testid="dropdown-body"
              id="popover-content-:r1:"
              role="dialog"
              style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.95) translateZ(0);"
              tabindex="-1"
            >
              <div
                class="chakra-popover__body css-0"
                id="popover-body-:r1:"
              >
                <ul
                  class="css-155za0w"
                  role="list"
                />
              </div>
            </section>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-57dgik"
    >
      <div
        class="chakra-stack css-1xum4w9"
      >
        <span
          class="css-ew0hq8"
        >
          <span
            class="css-zvlevn"
            data-testid="button-timeframe-1"
          >
            12m
          </span>
        </span>
        <span
          class="css-ew0hq8"
        >
          <span
            class="css-zvlevn"
            data-testid="button-timeframe-2"
          >
            6m
          </span>
        </span>
        <span
          class="css-ew0hq8"
        >
          <span
            class="css-zvlevn"
            data-testid="button-timeframe-3"
          >
            1m
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
